dev-haskell/cabal-install/metadata.xml

```diff
@@ -5,14 +5,20 @@
 		<email>haskell@gentoo.org</email>
```

Comment:
Should there be a commit keywording `cabal-install-3.2.0.0`? Its `metadata.xml` has been updated but not the ebuild itself.

It also appears that this `metadata.xml` change and a change in `cabal-3.2.1.0` adding `nocabaldep`, i.e. file changes from two separate packages, are part of the same commit. Usually, it's one package per commit. I strongly recommend using `repoman commit` rather than `git commit` to help enforce this (as in, don't use git commit where possible). You don't need to worry about git adding the right set of changes together. Just run `repoman commit` in the directory of the package and it'll handle things for you.

Reply:
@solpeth I'm afraid I'm a bit confused here. The last there commits in my PR are:

    1. [119A828](https://github.com/ezzieyguywuf/gentoo-haskell/commit/119a828bf38c45280df98281e24a5fc941610951), which bumps `dev-haskell/cabal` to`3.2.1.0`. This is a pre-requisite for `cabal-install-3.2.0.0`
    2. [de95cc8](https://github.com/ezzieyguywuf/gentoo-haskell/commit/de95cc87995f490f1e46646c13fa537147b3e28f), which keywords`hackage-security`. This package is *also* needed for `cabal-install-3.2.0.0`.
    3. Finally, [5b671a6](https://github.com/ezzieyguywuf/gentoo-haskell/commit/5b67a1636543876bf539561f7b62497372140b59) which finally bumps `cabal-install` to `3.2.0.0`

If this is unclear from the git commit messages please let me know and I can modify them.

Regarding the `metadata.xml` change: this came from my `hackport` command, which simply removed the `long description` (I think you may have commented on removing `long description` in IRC).

I reset the commit and ran `repoman commit` and still got the same commit as is listed here.

Can you elaborate a bit on how exactly you think this commit should look? I should note that the `metadata.xml` change actually occured in `Step 1` listed above, or two commits ago.

Reply:
The commit messages are fine :) it's what is contained in the commits (specifically your cabal-install commit). I don't see where it actually keywords `cabal-install-3.2.0.0` (your commit message actually says you bumped cabal-install, but there is no version to bump to because 3.2.0.0 is already in `::haskell`. So I have assumed you meant keywording it - like unmasking it. If this is incorrect let me know). If you check the `cabal-install-3.2.0.0` ebuild, I'm saying that the keywords will still be commented out. I suspect this is not what you were after.

Reply:
> 3.2.0.0 is already in `::haskell`

*facepalm* you are, of course, correct.

I got my rebasing mixed around again, and you're correct that my last commit here does not do what it is supposed to. Let me try to fix this and I'll re-push.

Reply:
Excellent! We understand each other :)

Reply:
I think I got it fixed, but let me know if it stills needs work.

I'll note: I used `repoman commit` again on that last commit, and it did include the `metadata.xml` changes.

Reply:
When I need to make a change to a package that I already should have, and it's not the latest commit (which would be reachable with a `git commit --amend --no-edit`, I do this:

1. Make the changes
2. `git add <file>`
3. git commit --fixup <commit hash of the commit I'm going to squash into, found from git log>
4. git rebase -i HEAD~<number of commits to go back to the one we're squashing into, e.g. 4> --autosquash
5. The interacting dialogue should show your `fixup` commit ready to be squashed into the commit you indicated previously. Close the dialogue and it will squash it in.

You can do multiple fixups and then squash them all at once, automatically, doing it like this. But it's certainly not the only way.

Reply:
@solpeth thanks for your thoughts.

The workflow I've been using is a bit different (as you said, there's many ways to do it):

1. `git rebase -i upstream/master`
2. Commit in which I should have made the change as `edit`
3. when the rebase stops, make the change I missed, commit as usual
4. Heck, I could do multiple commits if needed
5. Finally, `git rebase --continue`

I like to `rebase -i upstream master` each time as it gives me a good overview of what my entire PR will look like.